app/services/add_sandbox_template.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require_dependency "#{Rails.root}/lib/wiki_api"
+require_dependency "#{Rails.root}/lib/wiki_edits"
+
+class AddSandboxTemplate
+  def initialize(home_wiki:, sandbox:, sandbox_template:, current_user:)
+    @current_user = current_user
+    @sandbox = sandbox
+    @default_template = '{{user sandbox}}'
+    @sandbox_template = sandbox_template
+    @wiki_editor = WikiEdits.new(home_wiki)
+    @wiki_api = WikiApi.new(home_wiki)
+    @initial_page_content = @wiki_api.get_page_content(@sandbox)
+    add_template
+  end
+
+  private
+
+  def add_template
+    # Never double-post the sandbox template
+    return if sandbox_template_present?
+    default_template_present? ? replace_default_with_sandbox_template : add_sandbox_template
```

Comment:
There are three possibly paths here, depending on the state of the sandbox page: do nothing, add the template, replace the default template.

This will be clearer if all three paths are treated in parallel ways, either with a case statement or an if/elsif/else.

Reply:
Done!